### ISSUE TITLE
Unable to install with opam

### ISSUE THREAD

**Issue by rdavison** [owlbarn/eigen#13]
```
[ERROR] The compilation of eigen failed at "/Users/rdavison/.opam/opam-init/hooks/sandbox.sh build dune build eigen_cpp/libeigen_cpp_stubs.a".

#=== ERROR while compiling eigen.0.1.1 ========================================#
# context     2.0.3 | macos/x86_64 | ocaml-base-compiler.4.06.1 | https://opam.ocaml.org#52d5b52f
# path        ~/.opam/4.06.1/.opam-switch/build/eigen.0.1.1
# command     ~/.opam/opam-init/hooks/sandbox.sh build dune build eigen_cpp/libeigen_cpp_stubs.a
# exit-code   1
# env-file    ~/.opam/log/eigen-47600-9893d8.env
# output-file ~/.opam/log/eigen-47600-9893d8.out
### output ###
# [...]
# Called from file "bin/main.ml", line 170, characters 14-31
# Called from file "vendor/cmdliner/src/cmdliner_term.ml", line 27, characters 19-24
# Called from file "vendor/cmdliner/src/cmdliner.ml", line 106, characters 32-39
# Called from file "vendor/cmdliner/src/cmdliner.ml", line 136, characters 18-36
# Called from file "vendor/cmdliner/src/cmdliner.ml", line 251, characters 22-48
# Called from file "bin/main.ml", line 1219, characters 10-51
#
# I must not segfault.  Uncertainty is the mind-killer.  Exceptions are
# the little-death that brings total obliteration.  I will fully express
# my cases.  Execution will pass over me and through me.  And when it
# has gone past, I will unwind the stack along its path.  Where the
# cases are handled there will be nothing.  Only I will remain.
```

These two issues appear to be related:
- https://github.com/ocaml/opam/issues/3471
- https://github.com/ocaml/dune/issues/1039

tl;dr: Maybe need to add a `-p` flag to https://github.com/owlbarn/eigen/blob/master/eigen.opam#L10

**Comment by mseri** [comment]
That’s interesting, thanks for reporting and triaging it. Whith which version if dune it is happening?

**Comment by rdavison** [comment]
I was using dune 1.6.3.  However, as far as I can tell, it's supposedly a problem as far back as the earliest version of dune.  Furthermore, I kinda have a feeling that it has something to do with how opam2 does sandboxing.  I have sandboxing enabed on my laptop, but disabled in CI.  Besides that, and the fact that my laptop is a mac, and the CI environment is linux, I can't see any obvious differences.

**Comment by mseri** [comment]
It was just a curiosty. In my mac it did work fine but it is definitely more coherent with your fix in. Thanks again
